### DISCUSSION TITLE
Regarding maximum number of elements defined in PlanarArray

### DISCUSSION THREAD

**Discussion by ekilcioglu** [NVlabs/sionna#432] [Q&A]
Hello,
I am currently using Sionna RT to realize an indoor scenario with a transmitter, an RIS, and multiple receivers. I define the RIS as a PlanarArray with a specific number of rows and columns and then obtain the coverage map of it throughout the room by looking at a specific receiver position (I do it without considering the TX-RIS loss, just treating RIS as a different transmitter). My question is: What is the maximum number of dimensions that I can consider in Sionna RT? Is there a limit to the PlanarArray dimensions?

For my situation, I first realized 10x10 elements for the PlanarArray for the RIS and it worked for me to obtain the coverage map with num_samples=int(1.5e7), max_depth=0. However, when I want to define something like 20x20 dimensions, with the same num_samples and max_depth, the tool gives me the following error after a while during the coverage map calculation:

```
ResourceExhaustedError: {{function_node __wrapped____MklBatchMatMulV2_device_/job:localhost/replica:0/task:0/device:CPU:0}} OOM when allocating tensor with shape[2966720,400,3,1] and type float on /job:localhost/replica:0/task:0/device:CPU:0 by allocator mklcpu [Op:BatchMatMulV2] name: 
```

For your information, the indoor scenario that I consider is a 10x20 m room and I consider `cm_cell_size=[0.1,0.1]` while obtaining coverage maps. Let me know if you need additional information to answer this question.

Thank you so much for your answer in advance!

**Comment by SebastianCa** [comment]
Hi @ekilcioglu,

It’s difficult to provide detailed advice without knowing the exact implementation of the experiment. There is no strict limitation on the number of antenna elements; however, memory (and compute) requirements might limit your simulations. A 10x10 array sounds reasonable.

The error you are experiencing means that you are running out of memory. You either need to reduce the `num_samples` parameter or run the simulation on a system with more memory.

Remark: Why do you set max_depth to 0? Are you intentionally ignoring all reflections?
